File: informatika_ismeretek/kozep/2020_may/java/Feladvany.java
public class Feladvany {

    public final String feladvanyTeljes;
    public final int meret;

    public Feladvany(String sor) {
        this.feladvanyTeljes = sor;
        this.meret = (int) (Math.sqrt(sor.length()));
    }

    public void kirajzol() {
        for(var i = 0; i < feladvanyTeljes.length(); i++) {
            if(feladvanyTeljes.charAt(i) == '0') {
                System.out.print(".");
            }else{
                System.out.print(feladvanyTeljes.charAt(i));
            }
            if(i % meret == meret - 1) {
                System.out.println();
            }
        }
    }
}

File: informatika_ismeretek/kozep/2020_may/c#/Feladvany.cs
using System;

public class Feladvany {

    public readonly string feladvanyTeljes;
    public readonly int meret;

    public Feladvany(string sor) {
        this.feladvanyTeljes = sor;
        this.meret = Convert.ToInt32(Math.Sqrt(sor.Length));
    }

    public void Kirajzol() {
        for(var i = 0; i < feladvanyTeljes.Length; i++) {
            if(feladvanyTeljes[i] == '0') {
                Console.Write(".");
            }else{
                Console.Write(feladvanyTeljes[i]);
            }

            if(i % meret == meret - 1){
                Console.WriteLine();
            }
        }
    }
}